src/components/DataTable/DataTable.stories.tsx

```diff
@@ -24,7 +31,172 @@ export default {
     },
+    children: {
+      control: false,
+    },
+  },
+} as Meta<Args>;
+
+type Args = DataTableProps;
+
+type Person = {
+  firstName: string;
+  lastName: string;
+  age: number;
+  visits: number;
+  progress: number;
+};
+
+const defaultData: Person[] = [
+  {
+    firstName: 'Tanner',
+    lastName: 'Lindsey',
+    age: 24,
+    visits: 100,
+    progress: 50,
+  },
+  {
+    firstName: 'Tandy',
+    lastName: 'Miller',
+    age: 40,
+    visits: 40,
+    progress: 80,
+  },
+  {
+    firstName: 'Joe',
+    lastName: 'Dirte',
+    age: 45,
+    visits: 20,
+    progress: 10,
   },
+];
+
+const columnHelper = createColumnHelper<Person>();
+
+const columns = [
+  columnHelper.accessor('firstName', {
+    cell: (info) => info.getValue(),
+    header: () => <span>First Name</span>,
+    footer: (info) => info.column.id,
+  }),
+  columnHelper.accessor((row) => row.lastName, {
+    id: 'lastName',
+    cell: (info) => <i>{info.getValue()}</i>,
+    header: () => <span>Last Name</span>,
+    footer: (info) => info.column.id,
+  }),
+  columnHelper.accessor('age', {
+    header: () => 'Age',
+    cell: (info) => info.renderValue(),
+    footer: (info) => info.column.id,
+  }),
+  columnHelper.accessor('visits', {
+    header: () => <span>Visits</span>,
+    footer: (info) => info.column.id,
+  }),
+  columnHelper.accessor('progress', {
+    header: 'Profile Progress',
+    footer: (info) => info.column.id,
+  }),
+];
+
+export const Default: StoryObj<Args> = {
+  args: {},
+  render: (args) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [data] = React.useState(() => [...defaultData]);
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const table = useReactTable({
+      data,
+      columns,
+      getCoreRowModel: getCoreRowModel(),
+    });
+
+    const PersonDataTable = (args: DataTableProps<Person>) => {
+      // @ts-expect-error TODO-AH: what is the type issue here
+      return <DataTable {...args} />;
+    };
+
+    return <PersonDataTable {...args} table={table} />;
+  },
+};
+
+export const TableStyleBorder: StoryObj<Args> = {
+  args: {
+    tableStyle: 'border',
+  },
+  render: (args) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [data] = React.useState(() => [...defaultData]);
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const table = useReactTable({
+      data,
+      columns,
+      getCoreRowModel: getCoreRowModel(),
+    });
+
+    const PersonDataTable = (args: DataTableProps<Person>) => {
+      // @ts-expect-error TODO-AH: what is the type issue here
+      return <DataTable {...args} />;
+    };
+
+    return <PersonDataTable {...args} table={table} />;
+  },
+};
+
+export const TableSizeSm: StoryObj<Args> = {
+  args: {
+    tableStyle: 'border',
+    size: 'sm',
+  },
+  render: (args) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [data] = React.useState(() => [...defaultData]);
```

Comment:
If it's not being changed, why does the data need to be in state?

Reply:
This is just some boilerplate to be used later. We want to support selectable rows and sorting, which state will help with. I'm also going to factor these `render` calls so that they aren't so copy-paste